### PR TITLE
fixed treatment of default parameters

### PR DESCRIPTION
**Pull request by AlexanderSaydakov**

### CODE REVIEW

**Comment by mikeizbicki** [comment]
I just read trough your solution to setting the default parameters in C rather than in SQL.  I'm curious why you prefer the C version to the SQL version?  (I don't know much about writing C extensions for postgres, but I'd love to learn more.)  My initial impression is that the C version results in a bit more code duplication, but I'm wondering if it's maybe a bit faster than the SQL version?  I don't really know how postgres handles dynamic dispatch, but I could see default parameters in the SQL potentially having a bit more overhead.

**Comment by AlexanderSaydakov** [comment]
No particular reason. I am not sure about performance implications. I don't mind merging your PR as well if you change the default to 1.

**Comment by mikeizbicki** [comment]
Thanks.  I'll defer to your implementation and close the previous PR.

File: src/aod_sketch_pg_functions.c
Comment (jmalkin) [inline src/aod_sketch_pg_functions.c:176]: what exactly does `PG_NARGS()` return? This checks if the value is strictly greater than 1, but asks for value 1.

I'm blocked on reviewing this without knowing if that's reasonable.

Comment (AlexanderSaydakov) [reply src/aod_sketch_pg_functions.c:176]: It returns the number of arguments. So in this case if PG_NARGS() > 1 then there are at least two arguments. Therefore PG_GETARG_XXX(1) should be valid (zero-based index here).

**Review by jmalkin — APPROVED** [review]
Looks good.

Should we care about whether these are macros and to wrap the parts of the bincond in parens for safety? Or is that not a concern in practice here?

**Comment by AlexanderSaydakov** [comment]
Not sure what safety concern you are talking about. Maybe you are right in a general case, but this seems to work fine.

**Pull request merged**
